parla/tasks.py

```diff
@@ -447,6 +447,124 @@ def get_current_devices() -> List[Device]:
     return list(task_runtime.get_devices())
 
 
+@contextmanager
+def _reserve_persistent(memsize, device):
+    resource_pool = get_scheduler_context().scheduler._available_resources
+    resource_pool.allocate_resources(device, {'memory' : memsize}, blocking = True)
+    try:
+        yield
+    finally:
+        resource_pool.deallocate_resources(device, {'memory' : memsize})
+
+
+def _get_parla_device(device):
+    if isinstance(device, Device):
+        return device
+    try:
+        import cupy
+    except ImportError:
+        pass
+    else:
+        if isinstance(device, cupy.cuda.Device):
+            from .cuda import gpu
+            index = device.id
+            return gpu(index)
+    raise ValueError("Don't know how to convert object of type {} to a parla device object.".format(type(device)))
```

Comment:
This is OK for now. But please add a TODO that this needs to be made generic and moved to `device.py`. I suspect the implementation needs to be that `Architecture`s provide a method that attempts to convert an object to a Parla device, then a module level function tries all the `Architecture`s to find one that can convert the object, or fails. This could then be used anywhere a device is expected to provide Python style typing flexibility for device objects. All without having any reference to CUDA in the Parla core.

Reply:
Sounds good.